scripts/prepare-workspace.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Folders Location
+PWD="$(pwd)"
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ROOT_DIR="${DIR}/.."
+WORKSPACE_DIR="${ROOT_DIR}/workspace"
+META_JAVA_MASTER_DIR="${WORKSPACE_DIR}/meta-java-master"
+DOWNLOAD_DIR="${WORKSPACE_DIR}/download"
+BSP_DIR="${WORKSPACE_DIR}/bsp"
+BSP_DEVKIT_DIR="${BSP_DIR}/devkit"
+BSP_DEVKIT_EDISON_DIR="${BSP_DEVKIT_DIR}/edison"
+
+# Create workspace directory if missing
+if [ ! -d "${WORKSPACE_DIR}" ]; then mkdir ${WORKSPACE_DIR}; fi
+# Create download directory if missing
+if [ ! -d "${DOWNLOAD_DIR}" ]; then mkdir ${DOWNLOAD_DIR}; fi
+# Download and unzip meta-java-master patch if missing
+if [ ! -d "${META_JAVA_MASTER_DIR}" ]; then
+  if [ ! -d "${DOWNLOAD_DIR}/meta-java-master.zip" ]; then
+    cd ${DOWNLOAD_DIR}
+    wget http://git.yoctoproject.org/cgit/cgit.cgi/meta-java/snapshot/meta-java-master.zip;
+    cd ${PWD}
+  fi
+  unzip -qq -o ${DOWNLOAD_DIR}/meta-java-master.zip -d ${WORKSPACE_DIR};
+fi
+# Create bsp directory if missing
+if [ ! -d "${BSP_DIR}" ]; then mkdir ${BSP_DIR}; fi
+# Create devkit directory if missing
+if [ ! -d "${BSP_DEVKIT_DIR}" ]; then mkdir ${BSP_DEVKIT_DIR}; fi
+# Download and unzip iot-devkit src if missing
+if [ ! -d "${BSP_DEVKIT_EDISON_DIR}" ]; then
+  if [ ! -d "${DOWNLOAD_DIR}/iot-devkit-yp-poky-edison-20160606.zip" ]; then
+    cd ${DOWNLOAD_DIR}
+    wget http://iotdk.intel.com/src/3.5/edison/iot-devkit-yp-poky-edison-20160606.zip;
```

Comment:
that make sense for a sanity checking, but I was actually trying to do the actual build. Travis do not allow any job to run for more that 50 minutes, so I decided to do dry run in Travis, and create some build plans in Teamcity to do the actual build. And this needs to be done with the same sources, because I want to have a build environment that will work in CI and for a developer. So I will like to go with creating the actual build environment here. Do you have any guidance on that area?